### PR TITLE
update gorm dependencies

### PR DESCRIPTION
**Pull request by jimlambrt**
- [feature: validate resource the correct "kind"](https://github.com/hashicorp/go-dbw/commit/c84f8e05e5430418984bbd3364cc751be69bcad8)
- [chore: update gorm dependencies](https://github.com/hashicorp/go-dbw/commit/4fbe3c19df322dcaaaea09287320d1358bb1f4cb)
- [fix: ensure operations honor/use passed in context](https://github.com/hashicorp/go-dbw/commit/a7894b4f6846990bc48169815ee22982b5ba31a1)

### CODE REVIEW

File: lookup.go [hunk 1]
```diff
@@ -22,15 +21,15 @@ func (rw *RW) LookupBy(ctx context.Context, resourceWithIder interface{}, opt ..
...
+	if err := validateResourcesInterface(resourceWithIder); err != nil {
+		return fmt.Errorf("%s: %w", op, err)
 	}
 	where, keys, err := rw.primaryKeysWhere(ctx, resourceWithIder)
 	if err != nil {
 		return fmt.Errorf("%s: %w", op, err)
 	}
 	opts := GetOpts(opt...)
-	db := rw.underlying.wrapped
+	db := rw.underlying.wrapped.WithContext(ctx)
```
Comment (johanbrandhorst) [inline lookup.go:32]: Having to call `WithContext` on `wrapped` every time is a bit cumbersome and easy to forget, could we replace `.wrapped` with a function `getWrapped(ctx context.Context)` or something?

**Review by johanbrandhorst — COMMENTED** [review]
Ooh this is nice!

File: lookup.go (hunk 1 above)
Comment (jimlambrt) [reply lookup.go:32]: there are times we don't want it wrapped with a context.

**Review by johanbrandhorst — APPROVED** [review]

**Review by jefferai — COMMENTED** [review]
LGTM!

**Pull request merged**